File: OiWiki_Learning/快速幂/quickPow.java
public class quickPow{
    static long myQuickPow(long a,long b,long dev){
        //(a**p)%dev
        long ans=1;
        while(b!=0){
            if(b%2==1) ans=ans*a%dev;
            a=a*a%dev;
            b/=2;
        }
        return ans;
    }
    public static void main(String[] args){
        long dev=1000000007L;
        System.out.println(myQuickPow(2,100,dev));
    }
}

File: OiWiki_Learning/快速幂/quickPow.cpp
#include <stdlib.h>
#include <stdio.h>
using namespace std;
using ll=long long;
ll dev=1e9+7;
ll qpow(ll a, ll p,ll dev){
    //(a**p)%dev
    ll ans=1;
    while(p){
        if(p&1) ans=(ans*a)%dev;
        p/=2;
        a=(a*a)%dev;
    }
    return ans;
}
int main(){
    printf("%lld\n",qpow(2,100,dev));
    return 0;
}